lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -241,6 +241,7 @@ def add_3ds(post, options)
           post[:'3ds'][:attempt_n3d] = options[:attempt_n3d] if options[:attempt_n3d]
           post[:'3ds'][:challenge_indicator] = options[:challenge_indicator] if options[:challenge_indicator]
           post[:'3ds'][:exemption] = options[:exemption] if options[:exemption]
+          post[:'3ds'][:allow_upgrade] = options[:allow_upgrade] if options[:allow_upgrade]
```

Comment:
Looking at the [Checkout docs for this field](https://api-reference.checkout.com/previous/#operation/requestAPaymentOrPayout!path=0/3ds/allow_upgrade&t=request), it seems that the default value is `true` if nothing is specified in the request.

If that's the case, it seems like what we should be enabling here is the ability to send this field as `false`, but if options[:allow_upgrade] evaluates as `false`, this line won't be added to the request.

Reply:
So this field appears to be intended for use with the Gateway Specific/Direct 3DS only. The main thing I notice is on line 277 we have hard coded `post[:'3ds'][:enabled] = true` and in [their docs](https://api-reference.checkout.com/previous/#operation/requestAPaymentOrPayout) you can see the default is false and is set for `Whether to process this payment as a 3D Secure payment`. If it's always hard coded to true I'm not sure what passing in allow_upgrade is going to do. The transaction stays in a pending state so it's not possible to test the direct flow within a single remote test.